stripe/src/main/java/com/stripe/android/EphemeralKeyManager.java

```diff
@@ -54,18 +54,34 @@ TEphemeralKey getEphemeralKey() {
         return mEphemeralKey;
     }
 
+    @SuppressWarnings("checkstyle:IllegalCatch")
     private void updateKey(
             @NonNull String key,
             @Nullable String actionString,
             @Nullable Map<String, Object> arguments) {
+        // Key is coming from the user, so even if it's @NonNull annotated we
+        // want to double check it
+        if (key == null) {
+            mListener.onKeyError(HttpURLConnection.HTTP_INTERNAL_ERROR,
+                    "EphemeralKeyUpdateListener.onKeyUpdate was called " +
+                            "with a null value");
+        }
         try {
             mEphemeralKey = AbstractEphemeralKey.fromString(key, mEphemeralKeyClass);
+            mListener.onKeyUpdate(mEphemeralKey, actionString, arguments);
         } catch (JSONException e) {
             mListener.onKeyError(HttpURLConnection.HTTP_INTERNAL_ERROR,
-                    "The JSON from the key could not be parsed: "
-                            + e.getLocalizedMessage());
+                    "EphemeralKeyUpdateListener.onKeyUpdate was passed " +
+                            "a value that could not be JSON parsed: ["
+                            + e.getLocalizedMessage() + "] the raw body from Stripe's response" +
+                            " should be passed");
+        } catch (Exception e) {
```

Comment:
did you add this for safety?

Reply:
Yeah, if any other exception happens, the error will just be silenced, which was the case for a wrongly formatted JSON (the JSON is parsed, but then the `CustomerEphemeralKey` cannot be instanciated)